RejectedExecutionException

From Google Play for beta version 2017.02.28-RC:

User message:
Clicking on icons on the map

Samsung S7 with Android 7.0

```
io.reactivex.exceptions.UndeliverableException: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@50077de rejected from java.util.concurrent.ScheduledThreadPoolExecutor@d71ec00[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 12]
	at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:366)
	at io.reactivex.internal.schedulers.NewThreadWorker.scheduleActual(NewThreadWorker.java:134)
	at io.reactivex.internal.schedulers.NewThreadWorker.schedule(NewThreadWorker.java:51)
	at io.reactivex.Scheduler$Worker$PeriodicTask.run(Scheduler.java:400)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:59)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:51)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)
Caused by: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@50077de rejected from java.util.concurrent.ScheduledThreadPoolExecutor@d71ec00[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 12]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2049)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:814)
	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:305)
	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:550)
	at io.reactivex.internal.schedulers.NewThreadWorker.scheduleActual(NewThreadWorker.java:127)
	... 9 more
```

--- a/main/src/cgeo/geocaching/utils/AndroidRxUtils.java
+++ b/main/src/cgeo/geocaching/utils/AndroidRxUtils.java
@@ -8,7 +8,10 @@ import android.support.v4.app.Fragment;
 
 import java.lang.ref.WeakReference;
 import java.util.concurrent.Callable;
-import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 
 import io.reactivex.Maybe;
 import io.reactivex.Observable;
@@ -26,9 +29,11 @@ public class AndroidRxUtils {
 
     public static final Scheduler computationScheduler = Schedulers.computation();
 
-    public static final Scheduler networkScheduler = Schedulers.from(Executors.newFixedThreadPool(10, new RxThreadFactory("network-")));
+    private static final ThreadPoolExecutor.DiscardPolicy DISCARD_POLICY = new ThreadPoolExecutor.DiscardPolicy();
 
-    public static final Scheduler refreshScheduler = Schedulers.from(Executors.newFixedThreadPool(3, new RxThreadFactory("refresh-")));
+    public static final Scheduler networkScheduler = Schedulers.from(newFixedDiscardingThreadPool(10, "network-"));
+
+    public static final Scheduler refreshScheduler = Schedulers.from(newFixedDiscardingThreadPool(3, "refresh-"));
 
     private static final HandlerThread looperCallbacksThread =
             new HandlerThread("looper callbacks", Process.THREAD_PRIORITY_DEFAULT);
@@ -129,4 +134,17 @@ public class AndroidRxUtils {
             }
         });
     }
+
+    /**
+     * Provide an executor service with a fixed number of threads and an unbounded queue. If the
+     * service is shutdown while tasks are still queued, jobs will be silently discarded.
+     */
+    private static ExecutorService newFixedDiscardingThreadPool(final int nThreads, final String prefix) {
+        return new ThreadPoolExecutor(nThreads, nThreads,
+                0L, TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<Runnable>(),
+                new RxThreadFactory(prefix),
+                DISCARD_POLICY);
+    }
+
 }

